dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/enrollment/JdbcEnrollmentStore.java

```diff
@@ -0,0 +1,641 @@
+        }
+      }
+
+      return enrollment;
+    }
+
+    private Geometry mapGeometry(String geometry) {
+      if (Strings.isNullOrEmpty(geometry)) {
+        return null;
+      }
+
+      try {
+        WKBReader reader = new WKBReader();
+        byte[] bytes = WKBReader.hexToBytes(geometry);
+        return reader.read(bytes);
+      } catch (ParseException e) {
+        log.error("Error mapping enrollment geometry: {}", geometry);
+        return null;
+      }
+    }
+
+    private Sharing mapSharingJsonIntoSharingObject(String jsonSharing) {
+      if (StringUtils.isEmpty(jsonSharing)) {
+        return null;
+      }
+
+      try {
+        return JsonBinaryType.MAPPER
+            .readerFor(new TypeReference<Sharing>() {})
+            .readValue(jsonSharing);
+      } catch (IOException e) {
+        log.error("Error mapping enrollment sharing: {}", jsonSharing);
+        return null;
+      }
+    }
+
+    private UserInfoSnapshot mapUserInfo(String jsonUser) {
+      if (StringUtils.isEmpty(jsonUser)) {
+        return null;
+      }
+
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        return mapper.readValue(jsonUser, UserInfoSnapshot.class);
+      } catch (JsonProcessingException e) {
+        log.error("Error mapping enrollment user info: {}", jsonUser);
+        return null;
+      }
+    }
+
+    private List<Note> mapEnrollmentNotes(String jsonNotes) {
+      List<JdbcNote> jdbcNotes;
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        jdbcNotes = mapper.readValue(jsonNotes, new TypeReference<>() {});
+      } catch (JsonProcessingException e) {
+        log.error("Error mapping enrollment notes: {}", jsonNotes);
+        return List.of();
+      }
+
+      List<Note> notes = new ArrayList<>();
+      for (JdbcNote jdbcNote : jdbcNotes) {
+        Note note = new Note();
+        note.setUid(jdbcNote.getUid());
+        note.setNoteText(jdbcNote.getText());
+        note.setCreator(jdbcNote.getCreator());
+        note.setCreated(formatDate(jdbcNote.getCreated()));
+        User user = new User();
+        user.setUid(jdbcNote.getUpdatedByUid());
+        user.setUsername(jdbcNote.getUpdatedByUsername());
+        user.setFirstName(jdbcNote.getUpdatedByFirstname());
+        user.setSurname(jdbcNote.getUpdatedBySurname());
+        user.setName(jdbcNote.getUpdatedByName());
+        note.setLastUpdatedBy(user);
+        notes.add(note);
+      }
+
+      return notes;
+    }
+
+    private Set<TrackedEntityAttributeValue> mapTrackedEntityAttributeValues(
+        String jsonAttributes, TrackedEntity trackedEntity) {
+      List<JdbcAttribute> attributes;
+      ObjectMapper mapper = new ObjectMapper();
+      try {
+        attributes = mapper.readValue(jsonAttributes, new TypeReference<>() {});
+      } catch (JsonProcessingException e) {
+        log.error("Error mapping enrollment attributes: {}", jsonAttributes);
```

Comment:
When is this exception being thrown?
Can you reproduce it?
Swallowing this exception for attributes that the user explicitly asked for it is scary

Reply:
This exception would be thrown if the json is malformed and can't be deserialized to `JdbcAttribute`.
However, I think it should not happen, because we validate the payload before we persist attributes into the database, so the data we fetch should always be in a correct state, that's why I'm swallowing this exception.

Reply:
I am still not convinced that we should swallow this exception.
If user is asking for enrollment with attributes and he gets back enrollment with no attributes, he would think enrollment has no attributes unless he goes and look into the logs.
I would prefer the call to fail with a 500 explaining what is happening and explaining that if the user still needs the enrollment he can ask for it without the attributes.